esphome/components/http_request/http_request.cpp

```diff
@@ -24,7 +24,7 @@ void HttpRequestComponent::set_url(std::string url) {
   this->client_.setReuse(true);
 }
 
-void HttpRequestComponent::send() {
+void HttpRequestComponent::send(std::vector<HttpRequestResponseTrigger *> response_triggers) {
```

Comment:
```suggestion
void HttpRequestComponent::send(const std::vector<HttpRequestResponseTrigger *> &response_triggers) {
```

With the current version, the vector would be copied on each function call. By declaring it a `const` reference, only the reference is passed.